src/mesh.f90

```diff
@@ -77,31 +80,51 @@ module m_mesh
 
 contains
 
-  function mesh_init(dims_global, nproc_dir, L_global, &
-                     periodic_BC) result(mesh)
+  function mesh_init(dims_global, nproc_dir, L_global, BC_x, BC_y, BC_z) &
+    result(mesh)
     !! Completely initialise the mesh object.
     !! Upon initialisation the mesh object can be read-only and shouldn't be edited
     !! Takes as argument global information about the mesh like its length, number of cells and decomposition in each direction
     integer, dimension(3), intent(in) :: dims_global
     integer, dimension(3), intent(in) :: nproc_dir ! Number of proc in each direction
     real(dp), dimension(3), intent(in) :: L_global
-    logical, dimension(3), optional, intent(in) :: periodic_BC
+    character(len=*), dimension(2), intent(in) :: BC_x, BC_y, BC_z
     type(mesh_t) :: mesh
 
-    integer :: dir
+    character(len=20), dimension(3, 2) :: BC_all
+    logical :: is_first_domain, is_last_domain
+    integer :: dir, j
     integer :: ierr
 
     allocate (mesh%geo)
     allocate (mesh%par)
-    mesh%global_vert_dims(:) = dims_global
 
-    if (present(periodic_BC)) then
-      mesh%periodic_BC(:) = periodic_BC
-    else
-      ! Default to periodic BC
-      mesh%periodic_BC(:) = .true.
-    end if
+    BC_all(1, 1) = BC_x(1); BC_all(1, 2) = BC_x(2)
+    BC_all(2, 1) = BC_y(1); BC_all(2, 2) = BC_y(2)
+    BC_all(3, 1) = BC_z(1); BC_all(3, 2) = BC_z(2)
+    do dir = 1, 3
+      do j = 1, 2
+        select case (trim(BC_all(dir, j)))
+        case ('periodic')
+          mesh%BCs_global(dir, j) = BC_PERIODIC
```

Comment:
we could add a check that both BCs are periodic

Reply:
Done